logrotate/src/LogRotateHelper.ipp

```diff
@@ -116,25 +164,24 @@ namespace {
         if (!boost::filesystem::exists(dir_path)) return;
 
         for (boost::filesystem::directory_iterator itr(dir_path);itr != end_itr; ++itr) {
-            std::string current_file(itr->path().filename().c_str());
+            std::string current_file(itr->path().filename().string());
             long time = 0;
             if (getDateFromFileName(app_name, current_file, time)) {
                 files.insert(std::pair<long, std::string > (time, current_file));
             }
         }
        
         //delete old logs.
-        int logs_to_delete = files.size() - max_log_count;
+        ptrdiff_t logs_to_delete = files.size() - max_log_count;
```

Comment:
You had an int.
.size() returns a size_t and the max_log_count is unsigned int if I remember correctly.
So it could possibly overflow if max_log_count is say 10 and you only got 1 file.